### PR TITLE
API-6691 - Handle exceptions during stream processing esp. sending co…

### PR DESCRIPTION
**Pull request by AndySpaven**
…nfirmation

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsapi/controllers/WrappedNotificationController.scala [hunk 1]
```diff
@@ -69,6 +72,11 @@ class WrappedNotificationsController @Inject() (
           (
             for {
               _ <- ET.cond(request.version == "1", (), BadRequest(JsErrorResponse(ErrorCode.INVALID_REQUEST_PAYLOAD, "Message version is invalid")))
+              _ <- ET.cond(
+                     ifConfirmationUrlExistsItMustBeHttps(request.confirmationUrl),
+                     (),
+                     BadRequest(JsErrorResponse(ErrorCode.INVALID_REQUEST_PAYLOAD, "Confirmation URL must have https protocol"))
```
Comment (johnsgp) [inline app/uk/gov/hmrc/pushpullnotificationsapi/controllers/WrappedNotificationController.scala:78]: This doesn't sound like an `INVALID_REQUEST_PAYLOAD` but rather an invalid protocol, but I don't know which error codes are expected by the clients.

Comment (AndySpaven) [reply app/uk/gov/hmrc/pushpullnotificationsapi/controllers/WrappedNotificationController.scala:78]: That's a good point - we've been using this for several situations but actually now you mention it, it's not right.

Comment (AndySpaven) [reply app/uk/gov/hmrc/pushpullnotificationsapi/controllers/WrappedNotificationController.scala:78]: It came from the fact that the version must be 1 using this as it's error type.  I think we considered payload to be the 5 or 6 fields and not just the body of the notification.
In fact now I think about it.... I'm not sure it's wrong.

Comment (AndySpaven) [reply app/uk/gov/hmrc/pushpullnotificationsapi/controllers/WrappedNotificationController.scala:78]: There is no spec for this.

**Pull request merged**